tsconfig.json

```diff
@@ -0,0 +1,16 @@
+{
+  "compilerOptions": {
+    "strict": true,
+    "target": "es5",
+    "declaration": true,
+    "outDir": "dist",
+    "pretty": true,
+    "sourceMap": true,
+    "moduleResolution": "node",
+    "allowJs": false
```

Comment:
I don't know if it should be a blocker but I'd like to see a few more compilerOptions for strictness.

```
"noImplicitAny": true,
"noUnusedParameters": true,
"noUnusedLocals": true,
```

@briancavalier @raveclassic Thoughts?

Reply:
I’m in favor of being as strict as possible. Are any of those covered already by `strict: true`?

Reply:
`noImplicityAny` is included in the `"strict": true` rule. 
`noUnusedLocals` has proved to lead to terrible DX in my practice. I'd better set up eslint correctly. 
`noUnusedParameters` - I don't have any experience with this but I'm not sure if it will a good idea to enable it.

Reply:
Ahh yes, a few are. Here's the full list from the docs
```
noImplicitAny, --noImplicitThis, --alwaysStrict, --strictBindCallApply, --strictNullChecks, --strictFunctionTypes and --strictPropertyInitialization.
```

So maybe just `noUnusedParameters` and `noUnusedLocals`

Reply:
Ok, I've update the PR with these rules added to root `tsconfig.json`

Reply:
Great, thanks!